### PR TITLE
feat: LT-2724-view-open-licence-returns

### PR DESCRIPTION
**Pull request by rorysproule1**

### CODE REVIEW

File: compliance/serializers.py [hunk 1]
```diff
@@ -42,6 +44,14 @@ def get_status(self, instance):
             }
         return None
 
+    def get_open_licence_returns(self, instance):
+        queryset = OpenLicenceReturns.objects.filter(organisation_id=instance.organisation_id).order_by(
+            "-year", "-created_at"
+        )
+
+        if queryset.exists():
```
Comment (BenAndrew-Kainos) [inline compliance/serializers.py:52]: No if required (serializers handle empty lists just fine :) )

File: compliance/views.py [hunk 2]
```diff
@@ -121,7 +120,8 @@ class OpenLicenceReturnsView(ListAPIView):
     serializer_class = OpenLicenceReturnsListSerializer
 
     def get_queryset(self):
-        return OpenLicenceReturns.objects.all().order_by("-year", "-created_at")
+        organisation_id = get_request_user_organisation_id(self.request)
+        return OpenLicenceReturns.objects.filter(organisation_id=organisation_id).order_by("-year", "-created_at")
```
Comment (BenAndrew-Kainos) [inline compliance/views.py:124]: Please add a unit test to ensure that you can only see your orgs OpenLicenceReturns (let me know if you need a hand)

File: compliance/serializers.py [hunk 3]
```diff
@@ -42,6 +44,14 @@ def get_status(self, instance):
...
+            return OpenLicenceReturnsListSerializer(queryset, many=True).data
```
Comment (BenAndrew-Kainos) [inline compliance/serializers.py:53]: Please add a unit test to ensure this is in the data

File: compliance/serializers.py (hunk 1 above)
Comment (rorysproule1) [reply compliance/serializers.py:52]: removed

File: compliance/views.py (hunk 2 above)
Comment (rorysproule1) [reply compliance/views.py:124]: Added with you

File: compliance/serializers.py (hunk 3 above)
Comment (rorysproule1) [reply compliance/serializers.py]: Added test

File: compliance/tests/tests_get_compliance.py [hunk 4]
```diff
@@ -0,0 +1,46 @@
+from django.urls import reverse
+
+from cases.enums import CaseTypeReferenceEnum
+from compliance.tests.factories import ComplianceSiteCaseFactory, OpenLicenceReturnsFactory
+from test_helpers.clients import DataTestClient
+
+
+def _assert_response_data(self, response_data, compliance_case, open_licence_returns):
```
Comment (andytaylorkainos) [inline compliance/tests/tests_get_compliance.py:8]: 👍

File: compliance/tests/tests_get_compliance_licences.py [hunk 5]
```diff
@@ -0,0 +1,6 @@
+from test_helpers.clients import DataTestClient
+
+
+class GetComplianceLicencesTests(DataTestClient):
+    def test_get_compliance_licences(self):
+
```
Comment (andytaylorkainos) [inline compliance/tests/tests_get_compliance_licences.py:6]: something missing here?

File: compliance/views.py [hunk 6]
```diff
@@ -66,7 +65,7 @@ class ComplianceManageStatus(APIView):
 
     authentication_classes = (GovAuthentication,)
 
-    def put(self, request, pk):
+    def put(self, request, pk):  # TODO: Move to CaseStatus endpoint
```
Comment (andytaylorkainos) [inline compliance/views.py:68]: when is this TODO going to be done? can you mention the story you intend to do it with

File: compliance/tests/tests_get_compliance_licences.py (hunk 5 above)
Comment (rorysproule1) [reply compliance/tests/tests_get_compliance_licences.py:16]: Ye, just added a test to assert the licences linked to the compliance case

File: compliance/tests/tests_get_compliance_licences.py [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+        licence = self.create_licence(application, is_complete=True)
+
+        url = reverse("compliance:licences", kwargs={"pk": compliance_case.id})
+        response = self.client.get(url, **self.gov_headers)
+        response_data = response.json()["results"]
+
+        self.assertEqual(len(response_data), 1)
+        self.assertEqual(response_data[0]["id"], str(licence.application_id))
+        self.assertEqual(response_data[0]["reference_code"], licence.application.reference_code)
+        self.assertEqual(response_data[0]["status"]["key"], licence.application.status.status)
```
Comment (BenAndrew-Kainos) [inline compliance/tests/tests_get_compliance_licences.py:20]: Do you not need to check the compliance licence details?

Comment (rorysproule1) [reply compliance/tests/tests_get_compliance_licences.py:20]: Those are the only details we grab from the licence, clicking the reference code then brings you to the licence itself

File: compliance/views.py (hunk 6 above)
Comment (rorysproule1) [reply compliance/views.py:72]: Added, Mark will be bringing in the change with his story (LT-1122)

File: compliance/tests/tests_get_compliance_licences.py [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+from test_helpers.clients import DataTestClient
+
+
+def _assert_response_data(self, response_data, licence):
+    self.assertEqual(len(response_data), 1)
+    self.assertEqual(response_data[0]["id"], str(licence.application_id))
+    self.assertEqual(response_data[0]["reference_code"], licence.application.reference_code)
+    self.assertEqual(response_data[0]["status"]["key"], licence.application.status.status)
+    self.assertEqual(len(response_data[0]["flags"]), 2)
+    self.assertEqual(response_data[0]["flags"][0]["name"], "Item not verified")
```
Comment (BenAndrew-Kainos) [inline compliance/tests/tests_get_compliance_licences.py:16]: Please get the properties of the flags rather than hardcoding what is expected

**Pull request merged**